### PR TITLE
Add FEA_TRACE_SUPPORT flag as an option to enable tracing.

### PR DESCRIPTION
**Pull request by tommikas**
@jupe

### CODE REVIEW

File: mbed-trace/mbed_trace.h [hunk 1]
```diff
@@ -110,7 +110,7 @@ extern "C" {
 #define tr_cmdline(...)         mbed_tracef(TRACE_LEVEL_CMD,     TRACE_GROUP, __VA_ARGS__)   //!< Special print for cmdline. See more from TRACE_LEVEL_CMD -level
 
 /** Possible to skip all traces in compile time */
-#if defined(YOTTA_CFG_MBED_TRACE) || (defined(YOTTA_CFG) && !defined(NDEBUG))
+#if defined(FEA_TRACE_SUPPORT) || defined(YOTTA_CFG_MBED_TRACE) || (defined(YOTTA_CFG) && !defined(NDEBUG))
```
Comment (jupe) [inline mbed-trace/mbed_trace.h]: this part should be check by @SeppoTakalo and perhaps @kjbracey-arm as well

**Comment by jupe** [comment]
@kjbracey-arm @SeppoTakalo  could you check this out

**Comment by kjbracey-arm** [comment]
Seems reasonable to me - if mbed_trace.h is acting as a replacement for ns_trace.h, FEA_TRACE_SUPPORT is the macro we seem to be using, with HAVE_DEBUG being deprecated. There should indeed be some non-Yotta-branded selection method.

There are still people using HAVE_DEBUG, mind. Although I'm fine with losing that - the name is unhelpful.

**Comment by jupe** [comment]
lgtm but lets wait feedback from @SeppoTakalo before merging

**Comment by tommikas** [comment]
@SeppoTakalo Could you take a look at this please.

**Comment by SeppoTakalo** [comment]
+1
I'm ok with the change.
When we take the mbed-trace library into use, we can replace the HAVE_DEBUG flag with the FEA_TRACE_SUPPORT.

**Pull request merged**